superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -489,9 +489,9 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
     addDangerToast,
   );
   const isDynamic = (engine: string | undefined) =>
-    availableDbs?.databases.filter(
+    availableDbs?.databases?.find(
```

Comment:
null or undefined won't have a method called find, so I think this will still fail?